Jenkinsfile

```diff
@@ -11,109 +11,192 @@ pipeline {
+                    sh 'mvn clean install -DskipStatic=true -DskipTests=true -pl $POMFIX'
+                }
+            }
+        }
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID != null }
+                    expression { env.CHANGE_TARGET != null }
+                }
+            }
             // TODO CAL-296 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
             steps {
                 parallel(
-                    linux: {
-                        node('linux-large') {
-                            retry(3) {
-                                checkout scm
+                        linux: {
+                            node('linux-large') {
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
+                                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                                        sh 'mvn install -pl !$DOCS -DskipStatic=true -DskipTests=true -T 1C'
+                                        sh 'mvn clean install -B -T 1C -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
+                                        sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
+                                    }
+                                }
                             }
-                            timeout(time: 1, unit: 'HOURS') {
-                                withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings') {
-                                    sh 'mvn clean install -B -T 1C -pl !$ITESTS'
-                                    sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
+                        },
+                        windows: {
+                            node('proxmox-windows') {
+                                bat 'git config --system core.longpaths true'
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
```

Comment:
@oconnormi will 1 hour be enough? I've seen several PR builds go for over 3 hours depending on how many modules were impacted in the dependency tree calculated by GIB. Maybe that's just because of the static analysis etc...?